Automattic/woocommerce-payments
# Updates our `<Pill />` component to use WooCommerce Core's Pill component

Fixes #5872 

#### Changes proposed in this Pull Request









This PR updates our `client/components/pill` to use the `@woocommerce/components` `<Pill />` component.

See #5872 for more details on why this change was made. The TL;DR version is that as part of our Deposits UI project, we needed a new Deposit status pill. Instead of creating a whole new `<DepositsStatusPill />` component from scratch there are benefits to updating our existing WCPay Pill to use WooCommerce Core's Pill component.

This PR includes 3 main changes:
 1. Update the existing `client/components/pill` to return WC's Pill
    1. I also updated this file to TS in the process.
 2. Update the `.scss` of the Pill to bring back the main style of our WCPay pill (grey text and border) and add support for more colour options for these Pills.
 3. Update all existing `<Pill />` usages to return their original styling (most cases I'm bringing back the smaller padding)

#### Usage of new Pill

```
<Pill>Default</Pill>
<Pill type="primary" className="custom-styles">Primary</Pill>
<Pill type="success">Success</Pill>
<Pill type="alert">Alert</Pill>
<Pill type="danger">Danger</Pill>
<Pill type="light">Light</Pill>
```

![image](https://user-images.githubusercontent.com/2275145/228431096-0e2948d2-a549-4a69-a859-b4bbd0ca6569.png)

#### Updating existing uses of `<Pill>`

 - Pills used on the Settings > Payment Methods page

`develop`            |  `this branch`
:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/2275145/228433398-f2ff3c4c-56b7-40e5-89f0-23f2fc79cd0d.png)|![image](https://user-images.githubusercontent.com/2275145/228439108-70172585-e740-4145-a826-75165dbb780c.png)


The notable difference is `line-height` styles is a bit smaller with the new WC Core Pill.

 - Pill used for the "New" label on Fraud Protection

`develop`            |  `this branch`
:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/2275145/228436258-cc45cc58-684d-4ad8-a568-7716311e1125.png)|![image](https://user-images.githubusercontent.com/2275145/228436408-20c7932a-6240-4902-9f63-a15c90927733.png)

The notable difference is the WC Core Pill has a bit more padding. I don't think it's worth updating the padding here as it looks fine. Keen to get others opinions though :)


#### Testing instructions





* The main testing done on this PR is checking that existing Pill usage has not changed in a drastic way.

1. To see the Fraud Protection banner on the overview page, add `wcpay_fraud_protection_settings_active` = 1 in your wp-options table and view the Overview page.
2. Go into the Payments > Settings and check the pills used to show fees have not changed.
3. To simply test some of the other Pills modified the `components/payment-methods-list/payment-method.js` and set conditions like `upeCapabilityStatuses.PENDING_APPROVAL === status` to always pass.

![image](https://user-images.githubusercontent.com/2275145/228438451-14c1db1f-65e9-4ef5-8ee8-9b4e52561ff9.png)


-------------------


**Post merge**
